src/ui/Search.js

```diff
@@ -0,0 +1,148 @@
+import React, {Component} from 'react';
+import Modal from 'react-modal';
+import {UP, DOWN, ESC, ENTER, PGUP, PGDN, F3} from '../keycode';
+
+// TODO: Make sure to bind modal to your appElement (http://reactcommunity.org/react-modal/accessibility/)
+// Modal.setAppElement('#yourAppElement');
+
+export default class extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {showSearchModal: false, searchString: '', searchMode: 0};
+    this.searchBoxNode = null;
+  }
+
+  handleKeyModal = e => {
+    if (e.keyCode === ENTER || e.keyCode === ESC) { // enter or escape
+      this.setState({showSearchModal: false});
+
+      // NOTE(Oak): getActiveNode uses document.activeElement, but right now
+      // our focus is on the modal, so we use setTimeout to wait for the modal to close
+      // first
+      setTimeout(() => {
+        if (this.state.searchString === '') return;
+        this.props.searchModes[this.state.searchMode].initSearch(this.state.searchString);
+      });
+
+    } else if (e.keyCode === UP) {
+      if (this.state.searchMode > 0) {
+        this.setState((prevState => ({searchMode: prevState.searchMode - 1})));
```

Comment:
we should announce the new search mode to blind users.

Reply:
I don't think this is needed (in the new UI the equivalent is switching tab). The following is already read:

> Speak (C) category=nav "Entered dialog"
> Speak (Q) category=nav "Dialog Document Tab list with 2 items"
> Speak (Q) category=nav "Search by string"
> Speak (Q) category=nav "Tab Selected 1 of 2"

press right

> Speak (I) category=nav "Search by block"
> Speak (Q) category=nav "Tab Selected 2 of 2"